x/data/features/msg_anchor.feature

```diff
@@ -0,0 +1,47 @@
+Feature: MsgAnchor
+
+  # Note: see ./types_content_hash.feature for content hash validation
+
+  Scenario: an error is returned if sender is empty
+    Given the message
+    """
+    {}
+    """
+    When the message is validated
+    Then expect the error "empty address string is not allowed: invalid address"
+
+  Scenario: an error is returned if sender is not a bech32 address
+    Given the message
+    """
+    {
+      "sender": "foo"
+    }
+    """
+    When the message is validated
+    Then expect the error "decoding bech32 failed: invalid bech32 string length 3: invalid address"
+
+  Scenario: an error is returned if content hash is empty
```

Comment:
This points to an issue with how we do validation that @robert-zaremba has raised, specifically that error reporting depends on ordering. What if we validate content hash first? Then these tests will change which just feels wrong. We should actually return all errors and bundle them together with something like https://github.com/hashicorp/go-multierror. Not for this PR obviously but something we should track. This would ideally be integrated with the SDK's error package.

Reply:
Looks like this was started in #941.